### PR TITLE
Tests ulyanov 2016 loss

### PR DESCRIPTION
**Pull request by jbueltemeier**

### CODE REVIEW

File: tests/ulyanov_et_al_2016/test_loss.py [hunk 1]
```diff
@@ -1,32 +1,155 @@
...
+        with subtests.test(impl_params=impl_params):
+            op = loss.UlyanovEtAl2016FeatureReconstructionOperator(
+                encoder, impl_params=impl_params,
+            )
+            op.set_target_image(target_image)
+            actual = op(input_image)
+
+            score = mse_loss(input_enc, target_enc)
+
+            desired = score / extract_batch_size(input_enc) if impl_params else score
```
Comment (pmeier) [inline tests/ulyanov_et_al_2016/test_loss.py:32]: By default the `input_image` and `target_image` have `batch_size==1`. Thus if we don't increase it manually, this will work regardless of a "correct" implementation.

File: tests/ulyanov_et_al_2016/test_loss.py [hunk 2]
```diff
@@ -1,32 +1,155 @@
...
+                    score_weight = 1e0
+                else:
+                    score_weight = 1e3 if stylization else 1e0
+            else:
+                score_weight = 1e0
+            assert style_loss.score_weight == pytest.approx(score_weight)
+
+
+def test_ulyanov_et_al_2016_perceptual_loss(subtests):
+    for stylization in (True, False):
```
Comment (pmeier) [inline tests/ulyanov_et_al_2016/test_loss.py:136]: For now you can ignore all cases with `stylization is False`. We probably want to change the API since it leads to errors such as #20. The discussion of this is out of the scope of this PR though.

File: tests/ulyanov_et_al_2016/test_loss.py [hunk 3]
```diff
@@ -1,32 +1,147 @@
...
+from pystiche.loss import PerceptualLoss
 from pystiche_papers.ulyanov_et_al_2016 import loss
 
 
-def test_ulyanov_et_al_2016_content_loss(
-    subtests, multi_layer_encoder_with_layer, batch_image
+def test_UlyanovEtAl2016FeatureReconstructionOperator(
+    subtests, multi_layer_encoder_with_layer, target_image, input_image
 ):
+    input_image = torch.cat((input_image, input_image), 0)
```
Comment (pmeier) [inline tests/ulyanov_et_al_2016/test_loss.py:18]: Nit 
Nit

```suggestion
    from pystiche_papers.utils import batch_up_image
    input_image = batch_up_image(input_image, 2)
```

File: tests/ulyanov_et_al_2016/test_loss.py [hunk 4]
```diff
@@ -1,32 +1,147 @@
...
     multi_layer_encoder, layer = multi_layer_encoder_with_layer
     encoder = multi_layer_encoder.extract_encoder(layer)
-    target_image = batch_image
-    input_image = batch_image.flip((0,))
     target_enc = encoder(target_image)
```
Comment (pmeier) [inline tests/ulyanov_et_al_2016/test_loss.py:21]: I think we should do the same with `target_image` and another `batch_size` so we can be sure that it has no effect.

Comment (jbueltemeier) [reply tests/ulyanov_et_al_2016/test_loss.py:22]: What do you mean by different `batch_size`? Due to the `mse_loss` the `batch _size` of the `target_image` can only have the size 1 or the same size as the `input_image`.

Comment (pmeier) [reply tests/ulyanov_et_al_2016/test_loss.py:22]: Ah sorry, I thought we were testing the style loss, where the `batch_size` is irrelevant :facepalm:. In that case, go for the same `batch_size` as the `input_image`.
